shared/app/index.native.js

```diff
@@ -24,7 +24,8 @@ module.hot &&
       // We use global.devStore because module scope variables seem to be cleared
       // out after a hot reload. Wacky.
       console.log('updating route defs due to hot reload')
-      global.store.dispatch(setRouteDef(require('./routes').default))
+      const routes = require('./routes')
```

Comment:
I'm a bit fuzzy on how `require` works. If I have a running app in the login flow, I change the app routes and trigger HMR, will logging in pull in the old app route tree or the new one? Hmm...

Reply:
My experiments with HMR is that updating `routes.js` always leads to a full reload, and I'm not sure why. So it's moot, I guess.